bluemira/base/parameter_frame/_frame.py

```diff
@@ -222,6 +224,22 @@ def __str__(self) -> str:
 
 
 def _validate_parameter_field(field, member_type: Type) -> Tuple[Type, ...]:
+    if isinstance(member_type, str):
+        # https://peps.python.org/pep-0563/#resolving-type-hints-at-runtime
+        # but eval is dangerous, try and protect against wild evaluations
+        # https://nedbatchelder.com/blog/201206/eval_really_is_dangerous.html
+        member_type = member_type.replace("__import__", "").replace("_", "")
+        member_regex = re.search(r"^Parameter[\[].*[\]]$", member_type)
+
+        if not hasattr(member_regex, "string"):
+            raise TypeError(f"Field '{field}' does not have type Parameter.")
+        try:
+            member_type = eval(member_type, {"Parameter": Parameter}, {})  # noqa: S307
+        except NameError:
+            raise ParameterError(
+                f"Field {member_type} can't be evaluated safely, consider"
+                " not importing `from __future__ import annotations`"
+            )
```

Comment:
This shoud be looked at carefully, not sure if it is the right solution @hsaunders1904 to importing `from __future__ import annotations` in a file that defines a `ParameterFrame`

Reply:
So I've found this https://docs.python.org/3/whatsnew/3.7.html#pep-563-postponed-evaluation-of-annotations, that seems to suggest to get types at runtime, we can use [typing.get_type_hints()](https://docs.python.org/3/library/typing.html#typing.get_type_hints). 

Have we tried pulling the types out of the `dataclass` using `get_type_hints()`? Hopefully we can get them as objects rather than strings this way.

```python
frame_types = typing.get_type_hints(MyParameterFrameClass)
types = [frame_types[f.name] for f in dataclasses.fields(MyParameterFrameClass)]
```

Also, do you have MRE? I can't get it to go boom 💥. We should write a test that replicates the error as well.

```python
"""Try to cause a ParameterFrame error related to postponed evaluation of annotations"""

from __future__ import annotations

from dataclasses import dataclass

from bluemira.base.parameter_frame import Parameter, ParameterFrame


@dataclass
class PFrame(ParameterFrame):

    a: Parameter[float]
    b: Parameter[int]


d = {
    "a": {"value": 3.14, "unit": ""},
    "b": {"value": 1, "unit": ""},
}

f = PFrame.from_dict(d)
print(f)
```

Reply:
For me the above produces this stack trace:

```python
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[1], line 20
     12     b: Parameter[int]
     15 d = {
     16     "a": {"value": 3.14, "unit": ""},
     17     "b": {"value": 1, "unit": ""},
     18 }
---> 20 f = PFrame.from_dict(d)
     21 print(f)

File ~/Documents/bm2/bluemira/base/parameter_frame/_frame.py:116, in ParameterFrame.from_dict(cls, data, allow_unknown)
    113 except KeyError as e:
    114     raise ValueError(f"Data for parameter '{member}' not found.") from e
--> 116 value_type = _validate_parameter_field(
    117     member, cls.__dataclass_fields__[member].type
    118 )
    119 try:
    120     _validate_units(param_data, value_type)

File ~/Documents/bm2/bluemira/base/parameter_frame/_frame.py:228, in _validate_parameter_field(field, member_type)
    224 def _validate_parameter_field(field, member_type: Type) -> Tuple[Type, ...]:
    225     if (member_type is not Parameter) and (
    226         not hasattr(member_type, "__origin__") or member_type.__origin__ is not Parameter
    227     ):
--> 228         raise TypeError(f"Field '{field}' does not have type Parameter.")
    229     value_types = get_args(member_type)
    230     return value_types

TypeError: Field 'a' does not have type Parameter.
```

Reply:
but the `get_type_hints` snippet does seem to work, I'll get a test added

Reply:
Ok let me know what you think

Reply:
> For me the above produces this stack trace:

Oh yeah, I get that error now... Must have been doing something strange last time 🤔